src/main/java/com/sourcegraph/webhook/Dispatcher.java

```diff
@@ -69,23 +76,36 @@ public void run() {
         try {
             Response response = (Response) request.executeAndReturn((resp) -> resp);
             if (response.isSuccessful()) {
-                log.debug("Successfully dispatched webhook (" + serializer.getName() + ") data to URL: [" + hook.endpoint + "].");
+                log.debug("Successfully dispatched webhook (" + this.name + ") data to URL: [" + hook.endpoint + "].");
+                WebhookRegistry.storeError(this.hook, null);
                 return;
             }
+            String error = "Failed to dispatch webhook: ";
+            error += response.getStatusCode();
+            error += " ";
+            error += response.getStatusText();
+            log.warn(error);
+            WebhookRegistry.storeError(this.hook, error);
         } catch (ResponseException e) {
-            log.debug("Failed to dispatch webhook data (" + serializer.getName() + ") to URL: [" + hook.endpoint + "]:\n" + e);
+            WebhookRegistry.storeError(this.hook, "Failed: " + e.getMessage());
+            log.warn("Failed to dispatch webhook data (" + this.name + ") to URL: [" + hook.endpoint + "]:\n" + e);
```

Comment:
Can we combine this `log.warn` with the previous `log.warn` inside the `try { }`? This one also includes the `hook.endpoint`

Reply:
Hmm they actually have 2 different reasons why they are thrown, one non 2xx error, one an exception raised 🤔 
I don't think it would be much more readable when we wrap it inside an exception class that we would need to create 🤔  
But I think it looks cleaner now as we have the logging helper functions